Select the first track of an album will fill the playlist with that album

pascalgeo opened the pull request:
The default behavior of the Squeezebox Touch when browsing albums and selecting a track is to replace current playlist with the newly selected album. Currently Squeezer requests to first clear the playlist, which stops the music, and then select and add another album. Having an option that will replace current playing album with a whole new one would be nice.

More over, when scrolling through albums' list, it would be nice to highlight the album that contains the playing tracks, in the same way the currently played track is highlighted in the playlist.

bradfitz commented:
250 commits is too much to review. Also, development has moved elsewhere. This is no longer the home for the version on the Play store.

kaaholst committed:
Use ic_menu_alarm and change it's colour to white

kaaholst committed:
Don't use vector drawable for the undo bar

for now, until we can upgrade to AndroidX and fix it properly

kaaholst committed:
Prepare for release 1.5.3.

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Merge branch 'release-1.5.3'

kaaholst committed:
Merge branch 'release-1.5.3' into develop

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Upgrade gradle plugin

kaaholst committed:
Migrate to AndroidX

kaaholst committed:
disconnect in onTaskRemoved

kaaholst committed:
Call startForeground immediately after startForegroundService

to try to avoid
"Conext.startForegroundService did not then call Service.startForeground"
exceptions

kaaholst committed:
Artist - Album for notifications

Like https://developer.android.com/reference/androidx/media/app/NotificationCompat.MediaStyle.html
suggest

kaaholst committed:
Disconnect button in notification

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Restore disconnect icon

kaaholst committed:
Discopnnect button for pre lollipop notifications

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Prepare for release 1.5.4

kaaholst committed:
Add settings and advanced settings nodes

kaaholst committed:
Merge branch 'release-1.5.4'

kaaholst committed:
Merge branch 'release-1.5.4' into develop

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Home menu items follows display settings

kaaholst committed:
cleanup

kaaholst committed:
Add French translation

submitted by 2019 Olivier Sirol <czo@free.fr>

kaaholst committed:
Ignore lint ImpliedQuantity

which is revealed in the french translation

kaaholst committed:
Prepare for release 2.0.0

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Remove unsused resources

kaaholst committed:
Fix bug from remove resources

kaaholst committed:
Danish translations

kaaholst committed:
Vector drawables - WIP

kaaholst committed:
Undo button as vector drawable

kaaholst committed:
Vector drawables for pre API 22

kaaholst committed:
Use app:srcCompat

kaaholst committed:
Use drawableLeftCompat

kaaholst committed:
Fix android:gravity=center bug on pre API 22

The gravity property does not work for items within layer-list on
certain API levels.

See for example here:
https://stackoverflow.com/questions/42885410/vector-drawable-in-layer-list-on-older-android-versions/43601921
or here:
https://stackoverflow.com/questions/34831142/how-to-center-vector-drawable-in-layer-list-without-scaling

These example shows that the issue can be worked around using a verbose
setup in each drawable. Since all embedded slim icons share the same background
and inset of the foreground icon, it is instead implemented in code.

kaaholst committed:
Workaround a bug in WebView on appcompat 1.1.0 API 21

See:
https://issuetracker.google.com/issues/141132133

Use context.getApplicationContext().

kaaholst committed:
Don't use AppCompatDelegate.setCompatVectorFromResourcesEnabled

1) This feature is disabled per default as per:
https://developer.android.com/reference/android/support/v7/app/AppCompatDelegate.html#setcompatvectorfromresourcesenabled
2) Even with this enabled we still crash on API 21 when inflate a drawable
with a nested vector drawable.
3) We dont need it anymore as we are now generating nested drawables in
code (as the gravity property doesn't work on earlier android versions).

kaaholst committed:
cleanup

kaaholst committed:
Remove Util.setAlpha

It is not needed anymore, because we are now min api 14.

kaaholst committed:
Move theme settings to View menu

kaaholst committed:
Set list style in onResume

In case the preferred list style has been changed in a sub activity.

kaaholst committed:
Support menu status messages

kaaholst committed:
Fix register pages in onItemsReceived

kaaholst committed:
Upgrade gradle plugin

kaaholst committed:
Support choice items

as per:
http://wiki.slimdevices.com/index.php/SqueezeCenterSqueezePlayInterface#Choices_array_in_Do_action

kaaholst committed:
Do not post volume change the first time volume is

received for a player

kaaholst committed:
HttpUrlConnection follow redirect

kaaholst committed:
Move view settings to slim settings menu

kaaholst committed:
Remove unused server strings

kaaholst committed:
Embed server strings

and create a gradle task to extract strings from slimserver and
squeezeplay and put them in the app source.

kaaholst committed:
Grid layout now default for normal size screen

kaaholst committed:
Create android.yml

kaaholst committed:
Merge branch 'develop' into json

kaaholst committed:
Support Window icon

kaaholst committed:
Default theme is now DARK

kaaholst committed:
Don't show connect dialog on login completed

events

kaaholst committed:
Remove obsolete connection states

- LOGIN_STARTED
- LOGIN_COMPLETED

Login is now handled as part of the connection process, and returned as
a HTTP code.

kaaholst committed:
Move home menu to connection state

so it can be shared among activities

kaaholst committed:
Add missing home menu icons

as pngs for now until svgs are ready

kaaholst committed:
Use <item_fields>.<window_fields>.<icon_id|icon>

if icon is not specified in <item_fields>
See:
http://wiki.slimdevices.com/index.php/SqueezeCenterSqueezePlayInterface#.3Cwindow_fields.3E

and remove locally defined icons which are then not used.

Also apply ICON_TEXT window stylei, if window is not specified in item,
to make home menus follow display settings. This was lost when moving
home menu to connection state.

kaaholst committed:
Override server based window style to PLAY_LIST

for item type "playlist".

And top level window style icon_list

kaaholst committed:
Retain top level window when activity is recreated

i.e. a window which is applied in onItemsReceived.

kaaholst committed:
Use server based window styles

so we can distinguish between icon_list and home_menu and only override
the former to play_list for playlist type items.

kaaholst committed:
Show parent view

in fixed view at the top of the list.

This view contains:
- Name of the parent, may be overridden by a window title.
- Artwork of parent of the current list if it exists and the current
  view menu is text only.
- Context menu of the parent, if it exists.

kaaholst committed:
Properly retain UI state on activity recreate

also note an issue when context menu takes to long to load.

kaaholst committed:
Split multi line slim item text

First line is dislayed with primary text appearance style, and following
lines are displayed with the secondary text appearance style.

kaaholst committed:
Remove crashlytics

until we can migrate to firebase

kaaholst committed:
Remove google analytics

until we can migrate to firebase

kaaholst committed:
Upgrade gradle plugin

kaaholst committed:
Upgrade 3 party libraries

kaaholst committed:
Bump min SDK version to 19 - Android 4.4

First version which has java.nio.charset.StandardCharsets, which is
required by jetty-client, which is required by cometd, which is the
protocol we must use in order to support all features of the server.

Also bump target and compile SDK to version to the latest API level 29
(Android 10).

kaaholst committed:
Context menu icon is now the overflow menu icon

kaaholst committed:
Fix AlarmSettingsDialog on activity recreate

kaaholst committed:
Remove obsolete connect menu item and button

we now show the connect activity when disconnected, so they are no
longer necessary.

kaaholst committed:
Reconnect to mysqueezebox.com after registration

Also send all connection state releated (sticky-)events from
ConnectionState and limit handling of ConnectionStates to
NowPlayingFragment and start-/stop-foreground in SqueezeService.

kaaholst committed:
Discover http port of local servers

kaaholst committed:
Remove info icon from connection error message

kaaholst committed:
Make pixel density specific artwork for

home music library and settings menus.

To avoid out of memory errors on low memory devices and higher quality
imagery on high pixel density devices.

kaaholst committed:
Fix artwork background

kaaholst committed:
Implement checkbox field of item including

on/off actions

See
http://wiki.slimdevices.com/index.php/SBS_SqueezePlay_interface#.3Citem_fields.3E

kaaholst committed:
Parcel checkbox fields of Item

which was forgotten in the previous commit

kaaholst committed:
Implement radio field of item

kaaholst committed:
cleanup

kaaholst committed:
Implement slider items

kaaholst committed:
Vector drawables for av icons

and convert vectors to bitmaps for low api notifications, as it vectors
doesn't work for remoteviews (neither via xml nor code).

kaaholst committed:
Implement Input.initialText

kaaholst committed:
Switch from ContextMenu to PopupMenu

ContextMenus are not suitable for async load of its items.

Return from long press with an empty menu causes Android to continue
processing as if the item are selected.

kaaholst committed:
Input type time

and collect input for do-actions via a popup dialog

kaaholst committed:
Move AlarmSettingsDialog to itemlist package

kaaholst committed:
Disable context menu in NowPlayingFragment when

register on SN

kaaholst committed:
Enable extras menu

and add support for slideshow.

kaaholst committed:
Use runOnUiThread instead of our own Handler

kaaholst committed:
Radial Volume Panel

kaaholst committed:
VolumePanel and current playlist from Now Playing

kaaholst committed:
Don't show the item to go to players if in PlayersActivity

kaaholst committed:
VolumeBar reset timeout on touch movement

kaaholst committed:
Vector graphics

- ml_folder
- settings_audio
- settings_screen
- shuffle

kaaholst committed:
Merge branch 'develop' into radial_volume_panel

kaaholst committed:
Versioncode 72

kaaholst committed:
Versioncode 72

kaaholst committed:
Slide up now playing and current playlist activities

from the bottom.

and back down if the back button is pressed.

kaaholst committed:
Style Volume Panel

and adjust indicator size to current device.

kaaholst committed:
VolumeBar dont timeout when trscking touch

kaaholst committed:
Fix now playing layout

Add spacing around album artwork to lower density between components and
to create space for the volume and playlist buttons

kaaholst committed:
Merge branch 'release-2.0.0'

kaaholst committed:
Merge branch 'master' into develop

kaaholst committed:
Merge branch 'develop' into radial_volume_panel

kaaholst committed:
Option to easy switch between list and grid layouts

kaaholst committed:
Make sure the option to switch list layout reflects

whether the list can switch layout and the current layout.

kaaholst committed:
Fix bug in the previous commit

kaaholst committed:
Merge branch 'develop' into radial_volume_panel

kaaholst committed:
Ignore icon based showBrieflies in NowPlayingActivity

kaaholst committed:
Handle invalid server address

and switch to material input text fields in server_address_view

kaaholst committed:
Substitute FloatLabelLayout with material design

kaaholst committed:
NPE protect open context menu in NowPlayingFragment

kaaholst committed:
Enable crossfade on image cache fade in

so images with transparent parts do not leave the loading image visible.

kaaholst committed:
Move hint to TextInputLayout

kaaholst committed:
Add Host request-header field

as required by the HTTP 1.1 protocol

kaaholst committed:
Show next and play buttons in notification

for API levels >= LOLLIPOP.

Fix #628

kaaholst committed:
Cancel pending LMS requests in onPause

kaaholst committed:
Fix leaked window exception

Close any context menu which might be showing when an activity is
paused.

kaaholst committed:
Seperate list laypout settings for home menu items

and Slimbrowseitems

kaaholst committed:
Move theme settings to preferences

kaaholst committed:
Update README.md

kaaholst committed:
Prepare for release 2.0.1.

kaaholst committed:
Merge branch 'develop' into radial_volume_panel

kaaholst committed:
Merge branch 'release-2.0.1'

kaaholst committed:
Merge branch 'release-2.0.1' into develop

kaaholst committed:
Merge branch 'develop' into radial_volume_panel

kaaholst committed:
Upgrade gradle and build tools

kaaholst committed:
Merge branch 'develop' into radial_volume_panel

kaaholst committed:
Go back to list layout for normal size screen

kaaholst committed:
Merge branch 'radial_volume_panel' into develop

kaaholst committed:
Remove players and alarm menu items from ActionBar

kaaholst committed:
Slightly faster animations for now playoing and

current playlist activities

kaaholst committed:
Globalsearch shortcut in ActionBar

kaaholst committed:
Restore download framework

No functionality yet

kaaholst committed:
Support mapping of the go action and

Action.Window.isContextMenu

And simplify sokme context menu logic, which is no longer necesary,
because we do not use the build Android context menu.

kaaholst committed:
Fix the _quantity action parameter

Work around an issue in LMS which assumes the number of sub items equals itemsPerResponse.
This causes the playControl action of our initial request of one item, to not include the
"Play All" item, because it thinks there is only one item.

kaaholst committed:
Handle isContextMenu like squeezeplay

LMS may send isContextMenu in the itemParams, but this is ignored by squeezeplay, so we must do the same.

kaaholst committed:
Properly handle home menu vs browse item layouts

home_menu from LMS just means "hasImage", whereas we use it for Home Menu Items
http://wiki.slimdevices.com/index.php/HomeMenuItemsVersusSlimbrowseItems

kaaholst committed:
Show secondary parent text in header

kaaholst committed:
Swipe down to dismiss now playing screen

kaaholst committed:
Remove PARENT_ACTIVITY meta data from the manifest

it is only needed for API level 26 and below.

kaaholst committed:
Now playing and current playlist navigation

Enter now playing screen by tapping or swipe up on now playing fragment.
Exit now playing by swipe down on now playing screen or the action bar
home button.

Enter current playlist via button on lower left of now playing screen.
Exit current playlist via swipe down on header or the action bar home
button.

To indicate the function of the home button, spefically that it doesn't
navigate home, set a close icon for the current playlist and a down
arrow for the now playing screen.

kaaholst committed:
Use home icon as action bar home icon

instead of the Squeezer launcher icon

kaaholst committed:
Restore download url

kaaholst committed:
Merge branch 'develop' into download

kaaholst committed:
Move server strings to own files

kaaholst committed:
WIP - Implement player preferences

playtrackalbum and defeatDestructiveTouchToPlay

Note - changes to playpreferences are not reflected in the app. This is
because the player detection logic only checks if the same players are
present. This has to be refasctored and also take status messages into
account.

kaaholst committed:
Make sure events are posted at the intended rate

Remove asny messages of the same type prior to scheduling the new event.

Also remove a note about second-to-second updates, as we are not
subscribing to server event second-to-second anymore, but are doing the
calculations internal in the app.

kaaholst committed:
Replace the deprecated AsyncTask

Except ImageWorker for now

kaaholst committed:
Slightly increase timeout scanning for servers

to reduce the number of times we miss a server.

kaaholst committed:
Update player state and post event if

player preferences is updated via a serverstatus message.

kaaholst committed:
Fix option dialog with long texts

and hoist common code into BaseChoicesDialog and choices_layout

kaaholst committed:
Prepare for release 2.1.0.

kaaholst committed:
Mark untranslatable strings

kaaholst committed:
Update support info

in about dialog remove the Google group

kaaholst committed:
versioncode 75

kaaholst committed:
Image for button for input field follows app theme

kaaholst committed:
Refresh button for server scan follows app theme

kaaholst committed:
Fix warning from pre-launch report

Use icon_pending_artwork instead of icon_album_noart
Use standard paddings in material input fields

kaaholst committed:
Material themes

kaaholst committed:
Theme ActionBar

kaaholst committed:
Custom view for action bar

instead of the deprecated navigation mopde

kaaholst committed:
Elevated ActionBar for dark theme

kaaholst committed:
Merge branch 'develop' into download

kaaholst committed:
Upgrade gradle plugin

kaaholst committed:
Merge branch 'develop' into download

kaaholst committed:
color theming

kaaholst committed:
versionCode 82

kaaholst committed:
Theme volume panel

kaaholst committed:
getDimensionPixelSize instead of getDisplayMetrics

to calculate icon inset for item artwork

kaaholst committed:
Merge branch 'develop' into download

kaaholst committed:
Vector graphics and theme attributes

Added vector graphics for shuffle and repeat av buttons.
Use the colorCopntrolNormal theme attribute for av buttons

This eleminates the need for density and theme specific drwables, and
thus also the need for theme specific styles for av buttons.

kaaholst committed:
Victor graphics

ml_albums
ml_random
settings_repeat
settings_sleep
settings_sync

kaaholst committed:
Merge branch 'release-2.1.0'

kaaholst committed:
Different insets for item graphics

by embedding the inset into the vector grapchic and remove it from the
foreground of the layer drawable.

kaaholst committed:
Victor graphics

ml_playlist
internet_radio

kaaholst committed:
Merge branch 'release-2.1.0' into develop

kaaholst committed:
Merge branch 'develop' into download

kaaholst committed:
Handle global current playlist actions local

via the options menu and/or action bar.

kaaholst committed:
Download of current playlist items

kaaholst committed:
Remove test code to fetch menu items recursively

kaaholst committed:
Remove redundant substring parameter

kaaholst committed:
Move download command to Item

to prepare for download of different item types.

kaaholst committed:
Refactor to reflect we use SqueezePlay interface

Rename Plugin to JiveItem
Move SqueezePlay and Home menu fields to JiveItem
Move model objects to model

kaaholst committed:
Unify slim command handling in SlimCommand

idontusenumbers committed:
Added home screen widget; refactored ItemView/Activity to reuse on config screen

Fixed previous PR comments (missing icons in player selector, crashes due to threads messing with player state, etc), added multiple layouts, redid some of the icons, added button to activate squeezer to specific player

Added missing preview file and missing 2 column layout

Fixed crash when running on real hardware

Reconfigured remote button layout so it looks better on high density home screen; remote button uses styles now

Fixed empty player select list on widget config; fixed early service unbind; refactored remote service runner tasks

kaaholst committed:
Use icon_album_noart as default icon for now

playing screen

kaaholst committed:
Fix sw600dp/now_playing_fragment_mini

kaaholst committed:
Fix sw600dp/now_playing_fragment_mini

kaaholst committed:
Upgrade gradle plugin

kaaholst committed:
configure java 8 support

kaaholst committed:
Substitute some anonymous implementations with

lambda expressions or method references

kaaholst committed:
Configure multidex for api < 21

kaaholst committed:
Set volume panel background colors in code

on API < 21 you can't use attrs to color in the XML drawable

kaaholst committed:
Merge branch 'nik_develop' into develop

idontusenumbers committed:
Added preset buttons to widget

idontusenumbers committed:
Moved the remote button images to the java side; updated to upstream and incorporated some changes; converted some anonymous inner classes to lambdas

syphr42 committed:
Fix processing of partial JSON payloads from LMS

Fixes: #643

Signed-off-by: Gregory Moyer <moyerg@syphr.com>

idontusenumbers committed:
Refactored remote so buttons are customizable

Refactored remote so buttons are customizable

kaaholst committed:
Merge branch 'idontusenumbers-add-homescreen-widget' into develop

kaaholst committed:
Merge pull request #644 from syphr42/fix-partial-data-error

Fix processing of partial JSON payloads from LMS

kaaholst committed:
Move tint from vector drawable to style

Inflating a tinted vector drawable crashes api 19

kaaholst committed:
remove getTag

kaaholst committed:
Support download of myMusic

track, album, artist, genre, year, playlist and musicfolder

kaaholst committed:
Parcelable downloadAction so we can access it

form parent in sub acivity

sjalander committed:
Fixed minor typo of headings in privacy policy

kaaholst committed:
Support scoped storage for download

kaaholst committed:
Trottle download requests

to avoid possible problems with rate limit for notifications (via
download manager)

kaaholst committed:
Merge pull request #646 from sjalander/topic/privacy-policy

Fixed minor typo of headings in privacy policy

kaaholst committed:
Move common player menu items to playermenu.xml

and use them in both playerview and  main options menu.

kaaholst committed:
Cleanup SqueezeService.java

Fix compiler warnings
Replace anonymous with lambda
Remove obsolete build version check

kaaholst committed:
Cleanup NowPlayingFragment.java

Fix compiler warnings
Replace anonymous with lambda

kaaholst committed:
Support resizeableActivity

kaaholst committed:
Support Right to Left Layout

kaaholst committed:
Localized toasts for player not found for

home screen widgets

kaaholst committed:
Home screen widget auto connect to server

also wait for the sticky PlayersChanged event to handle the case where
no players are connected to ther server.

fixes #645

kaaholst committed:
Widget - Shortcuts for NowPlaying and current playlist

kaaholst committed:
Widget button select activity follows app theme

kaaholst committed:
widget - fix Danish translations

kaaholst committed:
Autoconnect if not manually disconnected

fix #639

kaaholst committed:
Larger cover art on Now Playing layout

kaaholst committed:
Prepare for release 2.2.0.

kaaholst committed:
versioncode 83

kaaholst committed:
Smaller cover art on Now Playing for small screens

to make sure all buttons are visible.

kaaholst committed:
Use RecyclerView.ViewHolder

kaaholst committed:
Merge branch 'release-2.2.0'

kaaholst committed:
Merge branch 'release-2.2.0' into develop

kaaholst committed:
Fix bug in Reflection.genericTypeResolver

so it can handle cases where the type arguments of the base class are
not mentioned in the type hierarchy up to the current class.

This is the case for PlayerBaseView.

kaaholst committed:
Change page size and remove the initial loading of

1 item, to workaround some issues in LMS where the page size is confused
with the number of items available

kaaholst committed:
Download confirmation and option to disable downloads

kaaholst committed:
Unselect other radio buttons

kaaholst committed:
Fix possible NPE in JiveItem.getTitlesParams

Skip play actions with "playlist control load ..." with a null id.
This can happen eg. when LMS returns an "Empty" item for a search without
any matching results.

kaaholst committed:
Current playlist control

Possibility to undo a clear current playlist
Remove a track by swiping left or right
Touch and hold to move a track

kaaholst committed:
Fix possible NPE showing async context menu

kaaholst committed:
Fix reported index error in OnItemSelectedListener

We have seen crashes from using parent.getItemAtPosition(pos) in
onItemSelected(AdapterView<?> parent, View view, int pos, long id)
so lookup the name at the selected position in the adapter instead.

kaaholst committed:
Fix possible NPE in OnSwipeListener

Apparently onFling can get a null e1 MotionEvent on some devices.

OnFling is rewritten to use velocityX and velocityY (wich can't be
null), to detect the direction of the swipe.

kaaholst committed:
Prepare for release 2.2.1.

kaaholst committed:
Upgrade gradle, build tools and dependencies

kaaholst committed:
Version code 85

kaaholst committed:
Upgrade play publisher plugin

kaaholst committed:
Use androidx PrerefenceManager

kaaholst committed:
Fix ClassCastException in CurrentPlaylistItemView.onIcon

if drawable of currently playing item is not a BitmapDrawable.

kaaholst committed:
Version code 87

kaaholst committed:
Add mime type when insert downloaded file into

the media store.

To avoid Android mistakenly assume it's an mp3 file.

kaaholst committed:
Merge branch 'release-2.2.1'
